src/Intl/Idn/Punycode/CodePointString.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Intl\Idn\Punycode;
+
+use ArrayIterator;
+use IteratorAggregate;
+
+/**
+ * Copyright (c) 2020 Trevor Rowbotham
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Originally forked from
+ * https://github.com/TRowbotham/punycode/tree/1.0.0
+ *
+ * @author Trevor Rowbotham <trevor.rowbotham@pm.me>
+ */
+class CodePointString implements IteratorAggregate
+{
+    /**
+     * @var array<int, int>
+     */
+    protected $codePoints;
+
+    /**
+     * @param string $input
+     */
+    public function __construct($input)
+    {
+        $this->codePoints = $this->utf8Decode($input);
+    }
+
+    /**
+     * @return \ArrayIterator<int, int>
+     */
+    public function getIterator()
+    {
+        return new ArrayIterator($this->codePoints);
```

Comment:
could it be worth replacing this class with an array? iterating over an array is always faster than using an iterator

Reply:
Done